doctr/models/modules/vision_transformer/pytorch.py

```diff
@@ -31,18 +29,25 @@ def __init__(
         self.num_patches = (height // patch_size[0]) * (width // patch_size[1])
 
         self.cls_token = nn.Parameter(torch.randn(1, 1, embed_dim))  # type: ignore[attr-defined]
-        self.positions = nn.Parameter(torch.zeros(1, self.num_patches + 1, embed_dim))  # type: ignore[attr-defined]
-        self.proj = nn.Conv2d(channels, embed_dim, kernel_size=self.patch_size, stride=self.patch_size)
+        self.positions = nn.Parameter(torch.randn(1, self.num_patches + 1, embed_dim))  # type: ignore[attr-defined]
+        self.proj = nn.Linear((channels * self.patch_size[0] * self.patch_size[1]), embed_dim)
 
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         B, C, H, W = x.shape
         assert H % self.patch_size[0] == 0, "Image height must be divisible by patch height"
         assert W % self.patch_size[1] == 0, "Image width must be divisible by patch width"
 
-        patches = self.proj(x)  # BCHW
+        # patchify image without convolution
+        # adopted from:
+        # https://uvadlc-notebooks.readthedocs.io/en/latest/tutorial_notebooks/tutorial15/Vision_Transformer.html
+        # NOTE: patchify with Conv2d works only with padding="valid" correctly on smaller images
+        # and has currently no ONNX support so we use this workaround
+        x = x.reshape(
+            B, C, (H // self.patch_size[0]), self.patch_size[0], (W // self.patch_size[1]), self.patch_size[1]
+        )
+        patches = x.permute(0, 2, 4, 1, 3, 5).flatten(1, 2).flatten(2, 4)  # (B, H, W, C, ph, pw) -> (B, H*W, C*ph*pw)
```

Comment:
```suggestion
        patches = x.permute(0, 2, 4, 1, 3, 5).flatten(1, 2).flatten(2, 4)  # (B, H', W', C, ph, pw) -> (B, H'*W', C*ph*pw)
```
I prefer to stick with their notations in the tutorial

Reply:
done